p2p/host/routed/routed_test.go

```diff
@@ -61,13 +55,36 @@ func TestRoutedHostConnectToObsoleteAddresses(t *testing.T) {
 
 	// Build routed host
 	rh := Wrap(h1, mr)
+	// Connection establishment should have worked without an error
+	require.NoError(t, rh.Connect(context.Background(), pi))
+	require.Equal(t, 1, mr.callCount, "the mocked FindPeer function should have been called")
+}
 
-	// Try to connect
-	err = rh.Connect(ctx, pi)
+func TestRoutedHostConnectFindPeerNoUsefulAddrs(t *testing.T) {
+	h1, err := basic.NewHost(swarmt.GenSwarm(t), nil)
+	require.NoError(t, err)
+	defer h1.Close()
 
-	// Connection establishment should have worked without an error
-	assert.NoError(t, err)
+	h2, err := basic.NewHost(swarmt.GenSwarm(t), nil)
+	require.NoError(t, err)
+	defer h2.Close()
+
+	// assemble the AddrInfo struct to use for the connection attempt
+	pi := peer.AddrInfo{
+		ID: h2.ID(),
+		// Use a wrong multi address for host 2, so that the initial connection attempt will fail
+		// (we have obsolete, old multi address information)
+		Addrs: []ma.Multiaddr{ma.StringCast("/ip4/127.0.0.1/tcp/1234")},
+	}
+
+	// Build mock routing module and replace the FindPeer function.
+	// Now, that function will return the correct multi addresses for host 2
+	// (we have fetched the most up-to-date data from the DHT)
+	mr := &mockRouting{findPeerFn: func(context.Context, peer.ID) (peer.AddrInfo, error) { return pi, nil }}
 
-	// The mocked FindPeer function should have been called
-	assert.Equal(t, 1, mr.callCount)
+	// Build routed host
+	rh := Wrap(h1, mr)
+	// Connection establishment should have worked without an error
+	require.Error(t, rh.Connect(context.Background(), pi))
```

Comment:
This comment and check doesn't make sense,
> // Connection establishment should have worked **without an error**

https://pkg.go.dev/github.com/stretchr/testify/require#Error
> Error asserts that a function returned **an error** (i.e. not `nil`). 

AFAIT you are testing that two back to back useless addresses do result in an error returned, so the comment is wrong and the code is correct.

Reply:
Yes, copy-pasted from the test above.